### DISCUSSION TITLE
How to execute `Unit test` for Packaged MSIX app in same project?

### DISCUSSION THREAD

**Discussion by shelllet** [microsoft/WindowsAppSDK#1362] [Q&A] [answered]
The tests in this test container can be executed only in windows store app unit test execution mode

![image](https://user-images.githubusercontent.com/16729017/131811601-2fddab45-ac8a-4257-952d-6a79ee5c4cbd.png)

**Comment by andrewleader** [comment, marked as answer]
Hey @Scottj1s or @alwu-msft can you help with this question about how to do unit testing in packaged MSIX?

**Reply by evelynwu-msft** [reply]
Unit testing requires use of Single-project MSIX Packaging. This [MSDN article](https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/single-project-msix?tabs=csharp-vs2019) describes how to create such a project, while this [blog post](https://devblogs.microsoft.com/ifdef-windows/winui-desktop-unit-tests/) walks through the process for adding unit tests (note that the blog post is several months old and so references Project Reunion 0.5.7, but the information contained therein is still applicable).

**Reply by shelllet** [reply]
Single-project MSIX is an experimental feature, [Try the new Single Project MSIX support! ](https://github.com/microsoft/WindowsAppSDK/discussions/1283), I encountered a lot of errors a few weeks ago, but this is a way。
